app/components/create/create-collection-dialog.tsx

```diff
@@ -31,27 +34,33 @@ export function CreateCollectionContent({
   collection,
   onCompleted,
 }: Pick<CreateCollectionDialogProps, "collection" | "onCompleted">) {
-  const completed = useRef(new Set<string>());
   const { progress, isSuccess, isError, queries } = useCreateCollectionQuery(
     collection.id,
     collection.items
   );
 
-  useEffect(() => {
-    if (isSuccess && !completed.current.has(collection.id)) {
-      const results = queries.map(
-        (query) => query?.data ?? { metadataUrl: "", imageUrl: "" }
+  const handlePublish = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      // TODO: Move the onCompleted logic that calls the program to this function
+      //       call onCompleted once we know the collection data is on chain...
+      onCompleted?.(
+        collection.id,
+        queries.map((query) => query?.data ?? { metadataUrl: "", imageUrl: "" })
       );
-      onCompleted?.(collection.id, results);
-      completed.current.add(collection.id);
-    }
-  }, [collection.id, isSuccess, onCompleted, queries]);
+    },
+    [collection.id, onCompleted, queries]
+  );
```

Comment:
@sxcamacho, here is the new spot; after the transaction succeeds, we can call onCompleted to close the dialog and move the collection from drafts to published in the data model in order to display it in the gallery.